### PR TITLE
New SLB resources.

### PR DESCRIPTION
**Pull request by devendra-bhatt-a10**
This PR contains one of my resource and rest are of Swapnil.

### CODE REVIEW

File: vthunder/resource_vthunder_slb_sip.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+func resourceSlbSipCreate(d *schema.ResourceData, meta interface{}) error {
+	logger := util.GetLoggerInstance()
+	client := meta.(vThunder)
+
+	if client.Host != "" {
+		logger.Println("[INFO] Creating SlbSip (Inside resourceSlbSipCreate) ")
+
+		data := dataToSlbSip(d)
+		logger.Println("[INFO] received V from method data to SlbSip --")
```
Comment (smitaraut-a10) [inline vthunder/resource_vthunder_slb_sip.go:45]: please make the logs more meaningful. Please replace V with better info.

File: vthunder/resource_vthunder_slb_sip.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+func resourceSlbSipRead(d *schema.ResourceData, meta interface{}) error {
+	logger := util.GetLoggerInstance()
+	client := meta.(vThunder)
+	logger.Println("[INFO] Reading SlbSip (Inside resourceSlbSipRead)")
+
+	if client.Host != "" {
+
+		data, err := go_vthunder.GetSlbSip(client.Token, client.Host)
+		if data == nil {
+
```
Comment (smitaraut-a10) [inline vthunder/resource_vthunder_slb_sip.go:64]: Add more logs here

File: vthunder/resource_vthunder_slb_sip_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package vthunder
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+var TEST_SLB_SIP_RESOURCE = `
+resource "vthunder_slb_sip" "testname" {
```
Comment (smitaraut-a10) [inline vthunder/resource_vthunder_slb_sip_test.go:10]: rename testname

File: vthunder/resource_vthunder_slb_sip_test.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+	sampling_enable {
+	    counters1 = "all"
+	}
+}
+`
+
+//Acceptance test
+func TestSlbSip_create(t *testing.T) {
```
Comment (smitaraut-a10) [inline vthunder/resource_vthunder_slb_sip_test.go]: Rename the function to start with TestAcc

File: vthunder/resource_vthunder_slb_smpp_test.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+var TEST_SLB_SMPP_RESOURCE = `
+resource "vthunder_slb_smpp" "testname" {
+	sampling_enable {
+	    counters1 = "all"
+	}
+}
+`
+
+//Acceptance test
+func TestSlbSmpp_create(t *testing.T) {
```
Comment (smitaraut-a10) [inline vthunder/resource_vthunder_slb_smpp_test.go]: rename the test function name

File: vthunder/resource_vthunder_slb_smpp_test.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
+package vthunder
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+var TEST_SLB_SMPP_RESOURCE = `
+resource "vthunder_slb_smpp" "testname" {
```
Comment (smitaraut-a10) [inline vthunder/resource_vthunder_slb_smpp_test.go:10]: rename the resource name

File: vthunder/resource_vthunder_slb_sip_test.go (hunk 4 above)
Comment (devendra-bhatt-a10) [reply vthunder/resource_vthunder_slb_sip_test.go]: This is already handled.

File: vthunder/resource_vthunder_slb_sip.go (hunk 1 above)
Comment (devendra-bhatt-a10) [reply vthunder/resource_vthunder_slb_sip.go:45]: Updated all resources with V in them.

File: vthunder/resource_vthunder_slb_dns.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+		if vc == nil {
+			logger.Println("[INFO] No dns found" + name)
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (devendra-bhatt-a10) [reply vthunder/resource_vthunder_slb_dns.go:80]: This err is been logged with API calling functions itself before returning.

File: website/docs/r/vthunder_slb_switch.html.markdown [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+layout: "vthunder"
+page_title: "vthunder: vthunder_slb_switch"
+sidebar_current: "docs-vthunder-slb-switch"
+description: |-
+    Provides details about vthunder SLB switch resource for A10
+---
+
+# vthunder\_slb\_switch
+
+`vthunder_slb_switch` Provides details about vthunder SLB transparent acl template
```
Comment (swapkashid) [inline website/docs/r/vthunder_slb_switch.html.markdown:11]: SLB transparent acl template need to be change to switch

**Pull request merged**